### PR TITLE
dataset-import script refactoring

### LINKED ISSUE #279: dataset-import script refactoring (pr/265)
### Description

Because of the new added features to this script, the initial structure doesn't respond to current needs (#265).

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by lhanusov**
fix #279

### CODE REVIEW

**Comment by ahus1** [comment]
@lhanusov - please rebase your PR once you continue working on this as there have been some changes for Infinispan in main.

**Comment by lhanusov** [comment]
@ahus1 will do thanks

**Comment by kami619** [comment]
@lhanusov thanks for making these changes Lukas, I am going through the changes and testing for any regressions, it might take a couple of days, depending on the other tasks I have at hand, please bear with me

File: doc/dataset/modules/ROOT/pages/using-provider.adoc [hunk 1]
```diff
@@ -24,28 +24,29 @@ The script assumes that the dataset provider is installed as described in xref:k
...
-5) create offline sessions in specific realm: -o | --create-offline-sessions '100 realm-0'
-6) delete specific realm/s with prefix -d | --delete-realms realm
-7) dataset provider status -s | --status
-8) dataset import script usage -h | --help
+Dataset import to the local minikube Keycloak application - usage:
+1) create realm/s with clients, users and password hash iterations - run -a (action) with or without other arguments: -a create-realms -r 10 -c 100 -u 100 -h 20000 -l 'https://keycloak.url.com'
+2) create clients in specific realm: -a create-clients -c 100 -n realm-0 -l 'https://keycloak.url.com'
+3) create users in specific realm: -a create-users -u 100 -n realm-0 -l 'https://keycloak.url.com'
+4) create events in specific realm: -a create-events -e 100 -n realm-0 -l 'https://keycloak.url.com'
+5) create offline sessions in specific realm: -a create-offline-sessions -o 100 -n realm-0' -l 'https://keycloak.url.com'
```
Comment (kami619) [inline doc/dataset/modules/ROOT/pages/using-provider.adoc:38]: there is a left over single quote next to `realm-0`

File: dataset/dataset-import.sh [hunk 2]
```diff
@@ -1,92 +1,130 @@
...
 dataset_provider_status_completed () {
-  KEYCLOAK_URI="${2:-"https://keycloak.$(minikube ip).nip.io"}/realms/master/dataset/"
   echo "Dataset provider status of the last completed task"
   t=0
-  STATUS_TIMEOUT=${1:-120}
-  until curl -ks "${KEYCLOAK_URI}status-completed" | grep '"success":"true"' ; do
-    if [[ $t -gt $STATUS_TIMEOUT ]]
+  until $(execute_command "status-completed" | grep '"success":"true"')
+  do
+    if [[ $t -gt $1 ]]
```
Comment (kami619) [inline dataset/dataset-import.sh:105]: @lhanusov I seem to see this behavior consistently where after a successful dataset job completion, running the `task dataset-import -- -a status-completed` gives an output of `../../dataset/dataset-import.sh: line 103: {"status":"previous: command not found` and able to consistently reproduce that, even though the status from the `status-completed` api seems to be coming back accurately.

**Full Log:**
```[kakella@kakella minikube]$ task dataset-import -- -a clear-status-completed
task: Task "split" is up to date
task: Task "ipchange" is up to date
task: Task "keycloak-cli-download" is up to date
task: Task "tlsdisableagent" is up to date
task: Task "env" is up to date
task: Task "prometheus" is up to date
task: Task "promtail" is up to date
task: Task "jaeger" is up to date
task: Task "tempo" is up to date
task: Task "loki" is up to date
task: Task "monitoring" is up to date
task: Task "dataset" is up to date
task: Task "datasetprovider" is up to date
task: Task "keycloak" is up to date
task: Task "keycloak-cli-unzip" is up to date
task: Task "gatlinguser" is up to date
[dataset-import] Action: [clear-status-completed] Dataset provider clears the status of the last completed task
[dataset-import] Skipping dataset status check
[dataset-import] 
[kakella@kakella minikube]$ task dataset-import -- -a create-offline-sessions -o 100 -n test-realm
task: Task "split" is up to date
task: Task "ipchange" is up to date
task: Task "keycloak-cli-download" is up to date
task: Task "tlsdisableagent" is up to date
task: Task "env" is up to date
task: Task "prometheus" is up to date
task: Task "promtail" is up to date
task: Task "tempo" is up to date
task: Task "loki" is up to date
task: Task "jaeger" is up to date
task: Task "dataset" is up to date
task: Task "monitoring" is up to date
task: Task "datasetprovider" is up to date
task: Task "keycloak" is up to date
task: Task "keycloak-cli-unzip" is up to date
task: Task "gatlinguser" is up to date
[dataset-import] Action: [create-offline-sessions] Creating 100 offline sessions in realm test-realm
[dataset-import] {"status":"Task started successfully","task":{"startTimeMs":"1673472778200","message":"Creation of 100 offline sessions"},"task-status-url":"https://keycloak.192.168.39.160.nip.io/realms/master/dataset/status"}
[kakella@kakella minikube]$ curl http://keycloak.$(minikube ip).nip.io/realms/master/dataset/status-completed
{"status":"previous task completed","task":{"success":"true","startTimeMs":"1673472778200","message":"Creation of 100 offline sessions","endTimeMs":"1673472779282"}}[katask dataset-import -- -a status-completed
task: Task "split" is up to date
task: Task "ipchange" is up to date
task: Task "tlsdisableagent" is up to date
task: Task "keycloak-cli-download" is up to date
task: Task "env" is up to date
task: Task "prometheus" is up to date
task: Task "tempo" is up to date
task: Task "loki" is up to date
task: Task "promtail" is up to date
task: Task "jaeger" is up to date
task: Task "monitoring" is up to date
task: Task "dataset" is up to date
task: Task "datasetprovider" is up to date
task: Task "keycloak" is up to date
task: Task "keycloak-cli-unzip" is up to date
task: Task "gatlinguser" is up to date
[dataset-import] Action: [status-completed] Dataset provider status of the last completed task
[dataset-import] ../../dataset/dataset-import.sh: line 103: {"status":"previous: command not found
[dataset-import] Polling...1s
[dataset-import] ../../dataset/dataset-import.sh: line 103: {"status":"previous: command not found
[dataset-import] Polling...2s
[dataset-import] ../../dataset/dataset-import.sh: line 103: {"status":"previous: command not found
^Ctask: Signal received: "interrupt"
```

File: dataset/dataset-import.sh [hunk 3]
```diff
@@ -1,92 +1,130 @@
...
-  curl -ks "${KEYCLOAK_URI}create-clients?count=$1&realm-name=$2"
+  execute_command "create-clients?count=$1&realm-name=$2"
 }
 
 create_users () {
-  KEYCLOAK_URI="${4:-"https://keycloak.$(minikube ip).nip.io"}/realms/master/dataset/"
   echo "Creating $1 user/s in realm $2"
-  check_dataset_status
-  curl -ks "${KEYCLOAK_URI}create-users?count=$1&realm-name=$2"
+  execute_command "create-users?count=$1&realm-name=$2"
```
Comment (kami619) [inline dataset/dataset-import.sh:77]: create-users seems to have regression originating from the DatasetResourceProvider.java, but I dont think its related to any changes here., this is merely an FYI. I will raise a different GH ticket for the issue I observed.

```
	
{"timestamp":"2023-01-11T21:29:42.844Z","sequence":5548,"loggerClassName":"org.jboss.logging.Logger","loggerName":"org.keycloak.benchmark.dataset.DatasetResourceProvider","level":"ERROR","message":"unable to complete task","threadName":"Thread-9","threadId":86,"mdc":{},"ndc":"","hostName":"keycloak-0","processName":"QuarkusEntryPoint","processId":1,"exception":{"refId":1,"exceptionType":"java.util.concurrent.ExecutionException","message":"java.lang.ArithmeticException: / by zero","frames":[{"class":"java.util.concurrent.FutureTask","method":"report","line":122},{"class":"java.util.concurrent.FutureTask","method":"get","line":191},{"class":"org.keycloak.benchmark.dataset.ExecutorHelper","method":"waitForAllToFinish","line":67},{"class":"org.keycloak.benchmark.dataset.DatasetResourceProvider","method":"createUsersImpl","line":393},{"class":"org.keycloak.benchmark.dataset.DatasetResourceProvider","method":"lambda$createUsers$10","line":362},{"class":"java.lang.Thread","method":"run","line":833}],"causedBy":{"exception":{"refId":2,"exceptionType":"java.lang.ArithmeticException","message":"/ by zero","frames":[{"class":"org.keycloak.benchmark.dataset.DatasetResourceProvider","method":"createUsers","line":999},{"class":"org.keycloak.benchmark.dataset.DatasetResourceProvider","method":"lambda$addUserCreationTasks$11","line":415},{"class":"org.keycloak.models.utils.KeycloakModelUtils","method":"lambda$runJobInTransaction$1","line":256},{"class":"org.keycloak.models.utils.KeycloakModelUtils","method":"runJobInTransactionWithResult","line":269},{"class":"org.keycloak.models.utils.KeycloakModelUtils","method":"runJobInTransaction","line":255},{"class":"org.keycloak.models.utils.KeycloakModelUtils","method":"runJobInTransactionWithTimeout","line":359},{"class":"org.keycloak.benchmark.dataset.ExecutorHelper","method":"lambda$addTaskRunningInTransaction$0","line":58},{"class":"java.util.concurrent.Executors$RunnableAdapter","method":"call","line":539},{"class":"java.util.concurrent.FutureTask","method":"run","line":264},{"class":"java.util.concurrent.ThreadPoolExecutor","method":"runWorker","line":1136},{"class":"java.util.concurrent.ThreadPoolExecutor$Worker","method":"run","line":635},{"class":"java.lang.Thread","method":"run","line":833}]}}}}
```

**Review by kami619 — COMMENTED** [review]
@lhanusov I have added my initial review comments, please have a look at them at your leisure, while I cover other areas of test.

File: dataset/dataset-import.sh (hunk 3 above)
Comment (lhanusov) [reply dataset/dataset-import.sh:77]: yes most probably caused by this commit: https://github.com/keycloak/keycloak-benchmark/commit/d6d55d4b017598e247597ec5d32cdb19351890ed

**Comment by lhanusov** [comment]
@kami619 thank you for your review, I hope it will help you to use it with better flexibility then before

**Review by kami619 — APPROVED** [review]
lgtm

**Pull request merged**